src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -214,7 +215,40 @@ public TextWriter EndWritingScope()
                 _originalWriter = null;
             }
 
-            return writer;
+            var tagHelperContentWrapperTextWriter = new TagHelperContentWrapperTextWriter(Encoding.UTF8);
+            var razorWriter = writer as RazorTextWriter;
+            if (razorWriter != null)
+            {
+                razorWriter.CopyTo(tagHelperContentWrapperTextWriter);
+            }
+            else
+            {
+                tagHelperContentWrapperTextWriter.Write(writer.ToString());
```

Comment:
What's the declared/runtime type of `writer` here? `TextWriter.ToString()` is not guaranteed to return the content - there can be trouble if the writer is replaceable.

We can leave this for a follow up as this assumption was present in the old code.